server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java

```diff
@@ -173,148 +176,170 @@ public String toString() {
-        response = with()
-                .param("access_token", "")
-                .get("/download/" + blobId);
+        response = Request.Get(
+                mainStepdefs.baseUri()
+                    .setPath("/download/" + blobId)
+                    .addParameter("access_token", "")
+                    .build())
+                .execute()
+                .returnResponse();
     }
 
     @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with a bad authentication token$")
-    public void getDownloadWithBadToken(String username, String attachmentId) {
+    public void getDownloadWithBadToken(String username, String attachmentId) throws ClientProtocolException, IOException, URISyntaxException {
         String blobId = blobIdByAttachmentId.get(attachmentId);
-        response = with()
-                .param("access_token", "bad")
-                .get("/download/" + blobId);
+        response = Request.Get(
+                mainStepdefs.baseUri()
+                    .setPath("/download/" + blobId)
+                    .addParameter("access_token", "bad")
+                    .build())
+                .execute()
+                .returnResponse();
     }
 
     @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with an invalid authentication token$")
-    public void getDownloadWithUnknownToken(String username, String attachmentId) {
+    public void getDownloadWithUnknownToken(String username, String attachmentId) throws ClientProtocolException, IOException, URISyntaxException {
         String blobId = blobIdByAttachmentId.get(attachmentId);
-        response = with()
-                .param("access_token", INVALID_ATTACHMENT_TOKEN)
-                .get("/download/" + blobId);
+        response = Request.Get(
+                mainStepdefs.baseUri()
+                    .setPath("/download/" + blobId)
+                    .addParameter("access_token", INVALID_ATTACHMENT_TOKEN)
+                    .build())
+                .execute()
+                .returnResponse();
     }
 
     @When("^\"([^\"]*)\" downloads \"([^\"]*)\" without blobId parameter$")
     public void getDownloadWithoutBlobId(String username, String attachmentId) throws Throwable {
         String blobId = blobIdByAttachmentId.get(attachmentId);
 
-        response = requestWithQueryParameterAuthentication(username, blobId)
-            .get("/download/");
+        URIBuilder uriBuilder = mainStepdefs.baseUri().setPath("/download/");
+        trustForBlobId(blobId, username);
+        AttachmentAccessTokenKey key = new AttachmentAccessTokenKey(username, blobId);
+        uriBuilder.addParameter("access_token", attachmentAccessTokens.get(key).serialize());
+        response = Request.Get(uriBuilder.build())
+            .execute()
+            .returnResponse();
     }
 
     @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with wrong blobId$")
     public void getDownloadWithWrongBlobId(String username, String attachmentId) throws Throwable {
```

Comment:
It's very messy but I think I changed the behavior in my changeset : one is supposed to test with an attachment access token and the other with an access token.